src/pages/workspace/reportFields/WorkspaceReportFieldsPage.tsx

```diff
@@ -125,39 +137,40 @@ function WorkspaceReportFieldsPage({
                         keyForList: String(reportField.fieldID),
                         orderWeight: reportField.orderWeight,
                         pendingAction: reportField.pendingAction,
-                        isSelected: selectedReportFields.find((selectedReportField) => selectedReportField.name === reportField.name) !== undefined && canSelectMultiple,
+                        isSelected: selectedReportFields.includes(getReportFieldKey(reportField.fieldID)) && canSelectMultiple,
                         isDisabled: reportField.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
                         text: reportField.name,
                         rightElement: <ListItemRightCaretWithLabel labelText={Str.recapitalize(translate(getReportFieldTypeTranslationKey(reportField.type)))} />,
                     })),
                 isDisabled: false,
             },
         ];
-    }, [filteredPolicyFieldList, policy, selectedReportFields, canSelectMultiple, translate]);
+    }, [filteredPolicyFieldList, policy, canSelectMultiple, translate, selectedReportFields]);
 
     useAutoTurnSelectionModeOffWhenHasNoActiveOption(reportFieldsSections.at(0)?.data ?? ([] as ListItem[]));
 
     const updateSelectedReportFields = (item: ReportFieldForList) => {
         const fieldKey = getReportFieldKey(item.fieldID);
-        const updatedReportFields = selectedReportFields.find((selectedReportField) => selectedReportField.name === item.value)
-            ? selectedReportFields.filter((selectedReportField) => selectedReportField.name !== item.value)
-            : [...selectedReportFields, filteredPolicyFieldList[fieldKey]];
-        setSelectedReportFields(updatedReportFields);
+        setSelectedReportFields((prevSelectedReportFields) => {
+            if (prevSelectedReportFields.includes(fieldKey)) {
+                return prevSelectedReportFields.filter((selectedField) => selectedField !== fieldKey);
+            }
+            return [...prevSelectedReportFields, fieldKey];
+        });
     };
 
     const toggleAllReportFields = () => {
         const availableReportFields = Object.values(filteredPolicyFieldList).filter((reportField) => reportField.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE);
-        setSelectedReportFields(selectedReportFields.length > 0 ? [] : availableReportFields);
+        setSelectedReportFields(selectedReportFields.length > 0 ? [] : Object.keys(availableReportFields));
```

Comment:
This is incorrect. The keys of `availableReportFields` are coming from `reportField.fieldID` where the selection array contains keys from `getReportFieldKey(reportField.fieldID)`

PS: This will be fixed after the removal of `selectionFieldList` and update `filteredPolicyFieldList`